src/main/java/org/sonar/plugins/csharp/sarif/SarifParser10.java

```diff
@@ -66,17 +69,24 @@ private static void handleIssue(JsonObject resultObj, SarifParserCallback callba
     String message = resultObj.get("message").getAsString();
 
     boolean hasLocation = false;
-    for (JsonElement locationEl : resultObj.get("locations").getAsJsonArray()) {
-      JsonObject locationObj = locationEl.getAsJsonObject();
-      if (locationObj.has("resultFile")) {
-        JsonObject resultFileObj = locationObj.get("resultFile").getAsJsonObject();
-        hasLocation = true;
-        String uri = resultFileObj.get("uri").getAsString();
-        String absolutePath = uriToAbsolutePath(uri);
-        JsonObject region = resultFileObj.get("region").getAsJsonObject();
-        int startLine = region.get("startLine").getAsInt();
-
-        callback.onIssue(ruleId, absolutePath, message, startLine);
+    if (resultObj.has("locations")) {
+      for (JsonElement locationEl : resultObj.get("locations").getAsJsonArray()) {
+        JsonObject locationObj = locationEl.getAsJsonObject();
+        if (locationObj.has("resultFile")) {
+          JsonObject resultFileObj = locationObj.get("resultFile").getAsJsonObject();
+          hasLocation = true;
+
+          if (!resultFileObj.has("uri") || !resultFileObj.has("region")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)